kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/bootstrap/FilterChainFactory.java

```diff
@@ -94,6 +95,11 @@ public ScheduledExecutorService eventLoop() {
                     return null;
                 }
 
+                @Override
+                public FilterDispatchExecutor filterDispatchExecutor() {
+                    return null;
```

Comment:
Maybe we should return a Stub impl that throws instead of `null`?

Reply:
Or make it throw with a message explaining the problem.

I think we should split it into FilterFactoryInitializationContext and FilterCreationContext, pretty sure this has been discussed before it's just another thing that would need to go through the deprecation/removal cycle

Reply:
yeah I think we agreed that in the past. Just wondered if we needed to repeat the pattern for the new impl...

Reply:
yeah I've pushed up making it throw instead, I think it's preferable to blow up ASAP, rather than returning a stub that then goes on to throw. It'd be more mysterious if we handed them a non-functional Executor I think. Have updated the javadoc on the interface so it describes the throw, rather than null return.